server/lib/realtime/rls/subscriptions/subscriptions.ex

```diff
@@ -0,0 +1,260 @@
+      with [_ | _] <- params_list,
+           [_ | _] = expected_users <-
+             Enum.reduce(params_list, [], fn
+               p, acc ->
+                 case Map.fetch(p, :user_id) do
+                   {:ok, user_id} -> [user_id | acc]
+                   :error -> acc
+                 end
+             end),
+           {:ok, %Postgrex.Result{rows: [_ | _] = rows}} <-
+             Repo.query(
+               "select u.id
+              from auth.users as u
+              join (
+                select *
+                from unnest($1::uuid[])
+                  as t(user_id)
+              ) as eu ON u.id = eu.user_id",
+               [expected_users]
+             ) do
+        {:ok, rows |> List.flatten() |> MapSet.new()}
+      else
+        _ -> {:ok, MapSet.new()}
+      end
+    end)
+  end
+
+  defp fetch_publication_tables(%Multi{} = multi) do
+    Multi.run(multi, :publication_entities, fn _, _ ->
+      Repo.query(
+        "select
+          schemaname,
+          tablename,
+          format('%I.%I', schemaname, tablename)::regclass as oid
+        from pg_publication_tables
+        where pubname = $1",
+        ["supabase_realtime"]
+      )
+      |> case do
+        {:ok,
+         %Postgrex.Result{
+           columns: ["schemaname", "tablename", "oid"] = columns,
+           num_rows: num_rows,
+           rows: rows
+         }}
+        when num_rows > 0 ->
+          publication_entities =
+            Enum.reduce(rows, %{}, fn row, acc ->
+              [{"schemaname", schema}, {"tablename", table}, {"oid", oid}] =
+                Enum.zip(columns, row)
+
+              acc
+              |> Map.put({schema, table}, [oid])
+              |> Map.update({schema}, [oid], fn oids -> [oid | oids] end)
+              |> Map.update({"*"}, [oid], fn oids -> [oid | oids] end)
+            end)
+
+          {:ok, publication_entities}
+
+        _ ->
+          {:ok, %{}}
+      end
+    end)
+  end
+
+  defp enrich_subscription_params(%Multi{} = multi) do
+    Multi.run(multi, :enriched_subscription_params, fn _,
+                                                       %{
+                                                         existing_users: existing_users,
+                                                         params_list: params_list,
+                                                         publication_entities: oids
+                                                       } ->
+      enriched_params =
+        case params_list do
+          [_ | _] ->
+            Enum.reduce(params_list, [], fn %{topic: topic, user_id: user_id}, acc ->
+              if MapSet.member?(existing_users, user_id) do
+                topic
+                |> String.split(":")
+                |> case do
+                  [schema, table, filters] ->
+                    filters
+                    |> String.split(~r/(\=|\.)/)
+                    |> case do
+                      [_, "eq", _] = filters ->
```

Comment:
we're only supporting "eq" (column equal to value) right now but will extend to others later